server/app/lib/frontend_analytics_tracer.rb

```diff
@@ -0,0 +1,13 @@
+module FrontendAnalyticsTracer
+  FRONTEND_LOGGER = Logger.new(File.join(Rails.root, 'log', 'frontend-queries.log'))
+  def analyze_query(query:)
+    params = {
+      user_ip: query.context[:request_ip],
```

Comment:
You asked in the PR

>I have the client rigged to send different HTTP headers for each but I don't know if I can access that at the level of the tracer?

You can access anything that you stick in the query context in the controller like we're doing with IP here.
So if you want to pull out your header in the controller, put it in the context, and then pull it out in one of your tracers, that will work!

Reply:
Oh.... yeah, duh